data-reconciliation-app/docs/adr/01-data-ingest.md

```diff
@@ -0,0 +1,139 @@
+# Data Ingest API
+
+## Status
+
+Proposed
+
+## Context
+
+We need to build an API that ingests member data into our system. Once ingested, we also need to store this data in the K-V store.
+
+Gathered requirements:
+
+- Members will upload data via csv
+- Members will upload multiple records (batch) or single records via csv
+- Data will have one attribute, and we will only work with string attributes for now.
+
+## Decision
+
+### Endpoint
+
+```
+POST  /csv
+// Endpoint ingests data into data reconciliation app
+	• URL Params
+        None
+	• Headers
+        Content-Type: application/json
+	• Request is ccfapp.Request object
+		○ Reuest Body = CSV text
+		○ Request UserID =
+	• Status: 200
+    • Status Text: "Data has been successfully recorded"
+```
+
+### Service Layer - CSV Service
+
+```
+// injects member data into CCF K-V store
+SubmitData(userId, csv body))
```

Comment:
Everything is data :) 

Given the service is called `csvService`,  this method could be `parse(userId, body): ServiceResult<value>` and the returning data can be sent to `Repository` to save into `K-V` store.